src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -340,6 +344,73 @@ class spell_pal_blessing_of_faith : public SpellScriptLoader
         }
 };
 
+// 190784 - Divine Steed
+class spell_pal_divine_steed : public SpellScriptLoader
+{
+public:
+    spell_pal_divine_steed() : SpellScriptLoader("spell_pal_divine_steed") { }
+
+    class spell_pal_divine_steed_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_pal_divine_steed_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STEED_HUMAN))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STEED_DRAENEI))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STEED_BLOODELF))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STEED_TAUREN))
+                return false;
+            return true;
+        }
+
+        void HandleOnCast()
+        {
+            Unit* caster = GetCaster();
+            if (!caster)
+                return;
+
+            uint32 spell_id = SPELL_PALADIN_DIVINE_STEED_HUMAN;
+            switch (caster->getRace())
+            {
+                case RACE_HUMAN:
+                    spell_id = SPELL_PALADIN_DIVINE_STEED_HUMAN;
+                    break;
+                case RACE_DWARF:
+                    spell_id = SPELL_PALADIN_DIVINE_STEED_HUMAN;
+                    break;
+                case RACE_DRAENEI:
+                    spell_id = SPELL_PALADIN_DIVINE_STEED_DRAENEI;
+                    break;
+                case RACE_BLOODELF:
+                    spell_id = SPELL_PALADIN_DIVINE_STEED_BLOODELF;
+                    break;
+                case RACE_TAUREN:
+                    spell_id = SPELL_PALADIN_DIVINE_STEED_TAUREN;
+                    break;
+                default:
+                    break;
+            }
+
+            if (spell_id)
```

Comment:
yeah it's bad, it was a prevention for crash but now it's not correct because spell_id = SPELL_PALADIN_DIVINE_STEED_HUMAN